pkg/controller/datadogagentdeployment/utils.go

```diff
@@ -104,32 +272,107 @@ func getInitContainers(logger logr.Logger, dad *datadoghqv1alpha1.DatadogAgentDe
 			Command:         []string{"bash", "-c"},
 			Args:            []string{"for script in $(find /etc/cont-init.d/ -type f -name '*.sh' | sort) ; do bash $script ; done"},
-			Env:             getEnvVarsForAgent(logger, dad),
+			Env:             envVars,
 			VolumeMounts:    volumeMounts,
 		},
 	}
+	if isSystemProbeEnabled(dad) {
+		systemProbeInit := corev1.Container{
+			Name:            "seccomp-setup",
+			Image:           spec.Agent.Image.Name,
+			ImagePullPolicy: *spec.Agent.Image.PullPolicy,
+			Resources:       *spec.Agent.Config.Resources,
+			Command: []string{
+				"cp",
+				"/etc/config/system-probe-seccomp.json",
+				"/host/var/lib/kubelet/seccomp/system-probe",
+			},
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      datadoghqv1alpha1.SystemProbeAgentSecurityVolumeName,
+					MountPath: "/etc/config",
+				},
+				{
+					Name:      datadoghqv1alpha1.SystemProbeSecCompRootVolumeName,
+					MountPath: "/host/var/lib/kubelet/seccomp",
+				},
+			},
+		}
+		containers = append(containers, systemProbeInit)
+	}
 
-	return containers
+	return containers, nil
 }
 
-// getEnvVarsForAgent converts Agent Config into container env vars
-func getEnvVarsForAgent(logger logr.Logger, dad *datadoghqv1alpha1.DatadogAgentDeployment) []corev1.EnvVar {
-	spec := dad.Spec
-	// Marshal tag fields
-	podLabelsAsTags, err := json.Marshal(spec.Agent.Config.PodLabelsAsTags)
-	if err != nil {
-		logger.Error(err, "failed to marshal pod labels as tags")
+// getEnvVarsForAPMAgent converts APM Agent Config into container env vars
+func getEnvVarsForAPMAgent(dad *datadoghqv1alpha1.DatadogAgentDeployment) ([]corev1.EnvVar, error) {
+	envVars := []corev1.EnvVar{
+		{
+			Name:  datadoghqv1alpha1.DDAPMEnabled,
+			Value: strconv.FormatBool(isAPMEnabled(dad)),
+		},
+		{
+			Name:  datadoghqv1alpha1.DDLogLevel,
+			Value: getLogLevel(dad),
+		},
 	}
-	podAnnotationsAsTags, err := json.Marshal(spec.Agent.Config.PodAnnotationsAsTags)
+	commonEnvVars, err := getEnvVarsCommon(dad)
 	if err != nil {
-		logger.Error(err, "failed to marshal pod annotations as tags")
+		return nil, err
 	}
-	tags, err := json.Marshal(spec.Agent.Config.Tags)
-	if err != nil {
-		logger.Error(err, "failed to marshal tags")
+	envVars = append(envVars, commonEnvVars...)
+	envVars = append(envVars, dad.Spec.Agent.Apm.Env...)
+	return envVars, nil
+}
+
+// getEnvVarsForProcessAgent converts Process Agent Config into container env vars
+func getEnvVarsForProcessAgent(dad *datadoghqv1alpha1.DatadogAgentDeployment) ([]corev1.EnvVar, error) {
+	envVars := []corev1.EnvVar{
+		{
+			Name:  datadoghqv1alpha1.DDProcessAgentEnabled,
+			Value: strconv.FormatBool(isProcessEnabled(dad)),
+		},
+		{
+			Name:  datadoghqv1alpha1.DDSystemProbeAgentEnabled,
+			Value: strconv.FormatBool(isSystemProbeEnabled(dad)),
+		},
+		{
+			Name:  datadoghqv1alpha1.DDLogLevel,
+			Value: getLogLevel(dad),
+		},
 	}
+	envVars = append(envVars, dad.Spec.Agent.Process.Env...)
```

Comment:
shouldn't we add append the common env vars as well?

Reply:
I prefer to have separated `Env[]` for each container

Reply:
I meant shouldn't we call `getEnvVarsCommon` as well?

Reply:
yes! I just saw it when testing it :)

Reply:
please make sure we add it to APM and system probe as well

Reply:
I think `DDLogLevel` should be in common env vars

Reply:
done